Source/JavaScriptCore/runtime/ArrayBuffer.h

```diff
@@ -400,6 +401,11 @@ void ArrayBuffer::unpin()
     m_pinCount--;
 }
 
+bool ArrayBuffer::isDetachable() const
+{
+    return !m_pinCount && !m_locked;
```

Comment:
Let's also check `isShared()` too. Shared one is also not-detachable.

Reply:
I'll correct this before merging, thank you